consumer/consumer.go

```diff
@@ -8,17 +8,17 @@ import (
 // IPConsumer processes IpRecord-s
 type IPConsumer interface {
 	// ConsumeIps processes all ip records from specified channel and specified client id
-	ConsumeIps(ctx context.Context, id string, ips <-chan goat_grpc.IpRecord) (DoneChannel, error)
+	ConsumeIps(ctx context.Context, id string, ips <-chan goat_grpc.IpRecord) (ResultsChannel, error)
```

Comment:
I agree it doesn't make too much sense without any context. If you check the `ConsumeIps` implementation, it should become more clear. The situation is, that there are some sanity checks done in the consumer before the processing itself begins - namely ensuring directory existence & template initialization. I thought the error in the return value could be used as an indication that the consumer failed to start (which is probably something I should add to the documentation now that I'm thinking about it) while ResultsChannel could be used to pass errors from the underlying goroutine. I'm not entirely sure this is the correct way to handle the situation because I wasn't able to find an example that would reflect all requirements for this situation.